include/net/net_ip.h

```diff
@@ -179,7 +179,10 @@ struct net_addr {
 extern const struct in6_addr in6addr_any;
```

Comment:
The commit message is way to long and describes things that are not relevant to this pr.
It should be enough to just say

```
POSIX defines INET_ADDRSTRLEN and INET6_ADDRSTRLEN as max sizes of
textual form of IP addresses (including terminating NUL):
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/netinet_in.h.html

We already define INET6_ADDRSTRLEN, so it makes sense to define
INET_ADDRSTRLEN too.
```

The rest of the story is irrelevant to this change.

Reply:
> The rest of the story is irrelevant to this change.

It's relevant, if you take into account that this patch not just introduces INET_ADDRSTRLEN, but tries to replace NET_IPV*_ADDR_LEN with INET*_ADDRSTRLEN.

But I get a hint from you and @ozhuraki's comments that adding it should be split from an attempt to switch to it. Will prepare a new PR then.